src/core/simulation.jl

```diff
@@ -89,17 +115,24 @@ function _check_chronologies(sim::Simulation)
 end
 
 function _assign_chronologies(sim::Simulation)
+    function find_val(d, value)
```

Comment:
More this function to utilities? @daniel-thom what do you think about this syntax?

Reply:
I'm fine with how this is.  It would be better as a utility function, such as `find_key_with_value`